lib/queue.rb

```diff
@@ -1,31 +1,102 @@
 class Queue
+# set queue_size here to something like 20 or whatever
+queue_size = 20
 
   def initialize
     # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    # raise NotImplementedError, "Not yet implemented"
+    # create a new array
+    @store = Array.new(queue_size)
+    #set the front and rear to position -1 bc it's empty still
+    @front = -1
+    @rear = -1
   end
 
+  #example: i want to add myself to a line of 20 people for coffee
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    # raise NotImplementedError, "Not yet implemented"
+    #check if front is empty
+    if @front == -1
+      #then rear is 1 and front is now 0
+      @rear = 1
+      @front = 0
+      #access the element in the store array
+      @store[@front] = element
+    #otherwise if front IS the rear, then the queue is full
+    elsif @front == @rear 
+      raise Error,"Tis Full"
+    #not empty
+    #calculate new rear then set it as the element
+    #set original rear to new rear
+    else
+      new_rear = (@rear + 1) % queue_size
+      @store[@rear] = element
+      @rear = new_rear
+    end
   end
 
+  #example: wth, this coffee line too long, i want to remove myself from this
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    # raise NotImplementedError, "Not yet implemented"
+    #check if front is empty, if yes, return nil
+    if @front == -1
+      return nil
+    #otherwise if front IS the rear, then the queue is full
+    elsif @front == @rear
+      raise Error, "Full up here"
+    #not empty
+    else 
+      #do same thing as the enqueue
+      new_front = (@front + 1) % queue_size
+      temp = @store[@front]
+      @store[@front] = nil
+      @front = new_front
+    end
+    return temp
   end
 
+  #the front is the front, so return the front
   def front
-    raise NotImplementedError, "Not yet implemented"
+    # raise NotImplementedError, "Not yet implemented"
+    return @front
   end
 
   def size
```

Comment:
You can use the positions of  front and rear to calculate the size of the queue.